app/federation/server/normalizers/room.js

```diff
@@ -0,0 +1,92 @@
+import { Federation } from '../index';
+
+const denormalizeRoom = (resource) => {
+	resource = { ...resource };
+
+	if (resource.t === 'd') {
+		resource.usernames = resource.usernames.map((u) => {
+			const [username, domain] = u.split('@');
+
+			return domain === Federation.domain ? username : u;
+		});
+	} else {
+		// Denormalize room name
+		const [roomName, roomDomain] = resource.name.split('@');
```

Comment:
what about creating a helper function to do this as it's done all over the place?